install.sh

```diff
@@ -1,9 +1,103 @@
+#!/bin/bash
 
+URL=https://github.com/LiqoTech/liqo.git
+FOLDER=liqo
+HELM_VERSION=v3.2.1
+HELM_URL=https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz
 
+function cleanup()
+{
+   export PATH=$OLD_PATH
+}
 
+# Define a string variable with a value
+commands="curl kubectl"
 
-curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
-chmod 700 get_helm.sh
-./get_helm.sh
+
+trap cleanup EXIT
+echo "[PRE-INSTALL]: Checking all pre-requisites are met"
+# Iterate the string variable using for loop
+for val in $commands; do
+    if command -v $val > /dev/null; then
+      echo "[PRE-INSTALL]: $val correctly installed"
+    else
+      echo "[PRE-INSTALL] [FATAL] : $val not found. Exiting"
+      exit 1
+    fi
+done
+
+set -e
+
+mkdir -p /tmp/liqo
+cd /tmp/liqo || exit 1
+mkdir -p /tmp/liqo
+cd /tmp/liqo || exit 1
+mkdir -p bin/
+cp linux-amd64/helm bin/
+export OLD_PATH=$PATH
+export PATH=$PATH:/tmp/liqo/bin/
+echo " [PRE-INSTALL] [HELM] Checking HELM installation..."
+HELM_INSTALLATION=false
+  if command -v helm > /dev/null; then
+    HELM_INSTALLED_VERSION=$(helm version --template "{{ .Version }}")
+    echo "[PRE-INSTALL]: Helm, found version: $HELM_INSTALLED_VERSION"
+    if [ "$HELM_INSTALLED_VERSION" != "$HELM_VERSION" ]; then
+         HELM_INSTALLATION=true
+         echo " [PRE-INSTALL] [HELM] found but having a potential incompatible version... Installing it"
+    fi
+  else
+    echo " [PRE-INSTALL] [HELM]: Helm not found... Installing it"
+    HELM_INSTALLATION=true
+fi
+if [ $HELM_INSTALLATION = true ]; then
```

Comment:
This looks to me unnecessarily complex. Just go and download your version even if it is already installed. It keeps your script simpler (you also do not need to edit `$PATH`); the probabilities to get your desired version would be anyway very low. Otherwise, insert `helm` as dependency at the beginning (optionally with a minimum version) and fail if it is not found.

For instance, consider that right now you are not using the correct version of helm in case already present, since the download folder is the last in `$PATH`